### PR TITLE
CORTX-30464: Add DPS features into PerfLine

### PR DESCRIPTION
**Pull request by maximalezhin**
## Problem Statement

This patch combines multiple DPS-related features that extend debugging capabilities of PerfLine:
- Single-script installation for K8s environment
- Interactive debugging sessions
- OS version detection (Rocky Linux / CentOS) during installation

Signed-off-by: Maxim Malezhin <maxim.malezhin@seagate.com>

## Design

-  Interactive script `/performance/PerfLine/env/install_dev_env.sh` allows to deploy K8s CORTX cluster and PerfLine utility by executing only a single script
- Interactive debugging feature for extended 

> **Following are the checklists for the author. Make sure all the items are considered. Write NA in front if any point is not applicable.**

## Coding

-   [x] Coding conventions are followed and code is consistent
-   [ ] No new Codacy Issues are added
-   [x] Version is updated in the respective VERSION file, format - `<tool>_v<major>.<minor>.<patch>`

## Documentation

-   [x] Changes done to readme file / Quick Start Guide
-   [ ] Docstrings are added to new functions/ files with Summary, Args and Return params

## Testing

-   [x] New/Affected tests are executed on Latest Build
-   [ ] Attach test execution logs
-   [x] No regression introduced

## Pull Request

-   [x] JIRA number/GitHub Issue added to PR, format - `<ID>: <Title>`
-   [x] Only one commit is present, format -
    ```txt
    <ID>: <Commit message>
    - bullets of changes committed in the PR
    ```
-   [x] Commit is sign-offed

## Review Checklist

-   [x] PR is self reviewed
-   [x] JIRA ticket state/status is updated
-   [x] Check if the DoD, description is clear and explained

## Impact Analysis

  (Checklist for Author/Reviewer/GateKeeper)

-   [x] Are there any changes in any common function?

  If yes,

-   [x] Relative/ dependent function calls are updated
-   [x] Executed all affected tests

> Note: If checkboxes are not working, write Y/N in front of the bullets. Thanks for your contribution and Happy Coding!

### CODE REVIEW

**Review by rahul27kumar — CHANGES_REQUESTED** [review]
Could please add readme files for dps? It would be also help for reviewer to review because it is a newly added feature for a perfline.

File: performance/PerfLine/env/install_dev_env.sh [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+SKIP_K8S=
+SKIP_DEPLOYMENT=
+SKIP_PERFLINE=
+
+K8S_HOSTS_FILENAME="hosts"
+CORTX_RE_DEFAULT_REPO="https://github.com/Seagate/cortx-re"
+CORTX_RE_DEFAULT_BRANCH="main"
+
+PERFLINE_DEFAULT_REPO="git@github.com:Seagate/seagate-tools.git"
+PERFLINE_DEFAULT_BRANCH="dps"
```
[the remark this answers is not in the document]
Comment (maximalezhin) [reply performance/PerfLine/env/install_dev_env.sh]: Done.

File: performance/PerfLine/env/solution.yaml [hunk 2]
```diff
@@ -0,0 +1,197 @@
+solution:
```
[the remark this answers is not in the document]
Comment (maximalezhin) [reply performance/PerfLine/env/solution.yaml]: Done.

**Comment by maximalezhin** [comment]
Requesting the next round of review. @rahul27kumar

File: performance/PerfLine/roles/perfline_setup/files/wrapper/continue_task.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
```
Comment (rahul27kumar) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/continue_task.sh:1]: Could you please add the usage of this functionality/feature on PerfLine README.md file? I didn't find any description/details about it.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/common_wait.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```
Comment (rahul27kumar) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/scripts/common_wait.sh:1]: same here. Add some steps and usage of this feature on README.md file of PerfLine.

File: performance/PerfLine/roles/perfline_setup/files/wrapper/wait_task.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
```
Comment (rahul27kumar) [inline performance/PerfLine/roles/perfline_setup/files/wrapper/wait_task.sh:1]: if this is also a part of new feature then please mention it on PerfLine README.md file.

**Review by rahul27kumar — CHANGES_REQUESTED** [review]
@maximalezhin  Please rename your readme.md file to README.md. your readme.md file helps me to review your PR. But still few features that you added as a part of this PR which is not mention anywhere in README.md file. Could you please have a look at my comment? And add usage/steps for continue_task, wait_task and common_wait.

File: performance/PerfLine/env/install_dev_env.sh [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+
+    user=$(yq_linux_386 ".specification.nodes[0].username" "$SPEC_FILE")
+    pass=$(yq_linux_386 ".specification.nodes[0].password" "$SPEC_FILE")
+
+    sed -i "s/^ansible_user=.*/ansible_user=$user/g" "$PERFLINE_HOSTS_FILENAME"
+    sed -i 's:^ha_type=.*:ha_type=hare:g' "$PERFLINE_HOSTS_FILENAME"
+    sed -i 's:^cortx_k8s_repo=.*:cortx_k8s_repo="/root/deploy-scripts":g' "$PERFLINE_HOSTS_FILENAME"
+    sed -i "s/^cluster_pass=.*/cluster_pass=$pass/g" "$PERFLINE_HOSTS_FILENAME"
+
+    sed -i 's:^disk=.*:disk="/dev/sdb":g' "$PERFLINE_HOSTS_FILENAME"
```
[the remark this answers is not in the document]
Comment (maximalezhin) [reply performance/PerfLine/env/install_dev_env.sh]: Agree

**Review by sampadap03 — APPROVED** [review]

**Review by rahul27kumar — APPROVED** [review]
Approved

**Comment by sampadap03** [comment]
Alex workflow issues are fixed. Mention of `host` is part of ansible design hence bypassing.

**Pull request merged**